### ISSUE TITLE
Transforming a Unit or Radial differential to a dimensional differential should do something about units

### ISSUE THREAD

**Issue by adrn** [astropy/astropy#6292]
As it stands:

```
>>> rep = coord.UnitSphericalRepresentation(lon=18*u.deg, lat=-11*u.deg)
>>> rep.represent_as(coord.SphericalRepresentation)
<SphericalRepresentation (lon, lat, distance) in (deg, deg, )
    ( 18., -11.,  1.)>
```

But for a differential
```
>>> rep = coord.SphericalRepresentation(lon=18*u.deg, lat=-11*u.deg, distance=10*u.pc)
>>> dif = coord.UnitSphericalDifferential(d_lon=15*u.mas/u.yr, d_lat=-11*u.mas/u.yr)
>>> dif.represent_as(coord.SphericalDifferential, rep)
<SphericalDifferential (d_lon, d_lat, d_distance) in (mas / yr, mas / yr, mas pc / (rad yr))
    ( 15., -11.,  -1.06581410e-14)>
```

At minimum, I think they should be consistent. So, the `d_distance` in the resulting `SphericalDifferential` could have dimensionless units.

This discussion was started at the end of #6289

cc @eteq @mhvk

**Comment by adrn** [comment]
Ahhhh keyboard went crazy...fixing.

**Comment by adrn** [comment]
Putting an optimistic 2.0 milestone on this since it's...kind of a bugfix?

**Comment by bsipocz** [comment]
@adrn - adding the "bug" label then, so if it doesn't make it to 2.0, it'll remilestoned to 2.0.1 rather than 3.0.0 :)

**Comment by mhvk** [comment]
@adrn - I'm not sure I agree; the `UnitSphericalDifferential` cannot know what the units on the radial part should be; arguably, given the distance in length units (remember this does not have to be a length for a representation), and the units of angle/time, it is logical to get out length/time. I'm slightly confused, though, why it isn't exactly zero.

Anyway, will try to think a bit more.

**Comment by adrn** [comment]
I don't quite follow, but I think I agree with you!

**Comment by adrn** [comment]
This would be solved by proposed changes in #6293, so closing this and adding this as a consideration to that issue.
